{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Evidence Export (from Elasticsearch)\n",
    "\n",
    "This will export ES records from the evidence data index into a parquet dataset (note that these records undergo several normalizations and modifications as well as scoring by comparison to the remote GS OT urls for evidence string files)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run utils.py\n",
    "import tqdm\n",
    "import os\n",
    "import os.path as osp\n",
    "import json\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import more_itertools\n",
    "from elasticsearch import helpers\n",
    "import pyarrow as pa\n",
    "export_dir = '/lab/data/raw/evidence-data-stage-2.parquet'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "es = get_es_client()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'_index': 'master_evidence-data',\n",
       " '_type': '_doc',\n",
       " '_id': '38c0851b3dba497d71ad645a01f498c3',\n",
       " '_score': 1.0,\n",
       " '_source': {'access_level': 'public',\n",
       "  'disease': {'biosample': {'id': 'http://purl.obolibrary.org/obo/UBERON_0002116',\n",
       "    'name': 'ileum'},\n",
       "   'efo_info': {'efo_id': 'http://www.ebi.ac.uk/efo/EFO_0000384',\n",
       "    'label': \"Crohn's disease\",\n",
       "    'path': [['EFO_0000408', 'EFO_0000405', 'EFO_0003767', 'EFO_0000384'],\n",
       "     ['EFO_0000408',\n",
       "      'EFO_0000540',\n",
       "      'EFO_0005140',\n",
       "      'EFO_0003767',\n",
       "      'EFO_0000384']],\n",
       "    'therapeutic_area': {'codes': [], 'labels': []}},\n",
       "   'id': 'EFO_0000384'},\n",
       "  'evidence': {'comparison_name': \"'Crohn's disease' vs 'non inflammatory bowel disease control' in 'A1b Paris classification'\",\n",
       "   'confidence_level': 'medium',\n",
       "   'date_asserted': '2019-05-14T15:19:25Z',\n",
       "   'evidence_codes': ['ECO_0000295'],\n",
       "   'evidence_codes_info': [[{'eco_id': 'ECO_0000295',\n",
       "      'label': 'RNA-seq evidence'}]],\n",
       "   'experiment_overview': \"RNA-seq of ileal biopsies from diagnostic endoscopy of pediatric Crohn's disease patients and non inflammatory bowel disease controls\",\n",
       "   'is_associated': True,\n",
       "   'log2_fold_change': {'percentile_rank': 99, 'value': 2.1},\n",
       "   'organism_part': 'http://purl.obolibrary.org/obo/UBERON_0002116',\n",
       "   'provenance_type': {'database': {'id': 'Expression_Atlas',\n",
       "     'version': 'prod.30'}},\n",
       "   'reference_replicates_n': 37,\n",
       "   'reference_sample': 'non inflammatory bowel disease control; A1b Paris classification',\n",
       "   'resource_score': {'method': {'description': 'DESeq2 version: 1.10.1'},\n",
       "    'type': 'pvalue',\n",
       "    'value': 3.64e-06},\n",
       "   'test_replicates_n': 193,\n",
       "   'test_sample': \"Crohn's disease; A1b Paris classification\",\n",
       "   'unique_experiment_reference': 'STUDYID_E-GEOD-101794',\n",
       "   'urls': [{'nice_name': 'ArrayExpress Experiment overview',\n",
       "     'url': 'http://identifiers.org/arrayexpress/E-GEOD-101794'},\n",
       "    {'nice_name': 'Gene expression in Expression Atlas',\n",
       "     'url': 'http://www.ebi.ac.uk/gxa/experiments/E-GEOD-101794?geneQuery=ENSG00000134438'},\n",
       "    {'nice_name': 'Baseline gene expression in Expression Atlas',\n",
       "     'url': 'http://www.ebi.ac.uk/gxa/genes/ENSG00000134438'}]},\n",
       "  'id': '38c0851b3dba497d71ad645a01f498c3',\n",
       "  'private': {'datasource': 'expression_atlas',\n",
       "   'datatype': 'rna_expression',\n",
       "   'eco_codes': ['ECO_0000295'],\n",
       "   'efo_codes': ['EFO_0000405',\n",
       "    'EFO_0000540',\n",
       "    'EFO_0005140',\n",
       "    'EFO_0000384',\n",
       "    'EFO_0000408',\n",
       "    'EFO_0003767'],\n",
       "   'facets': {'go': {'biological_process': [{'code': 'GO:0043010',\n",
       "       'term': 'camera-type eye development'},\n",
       "      {'code': 'GO:0021854', 'term': 'hypothalamus development'},\n",
       "      {'code': 'GO:0060173', 'term': 'limb development'},\n",
       "      {'code': 'GO:0007389', 'term': 'pattern specification process'},\n",
       "      {'code': 'GO:0045944',\n",
       "       'term': 'positive regulation of transcription by RNA polymerase II'},\n",
       "      {'code': 'GO:0007601', 'term': 'visual perception'}],\n",
       "     'cellular_component': [{'code': 'GO:0000790',\n",
       "       'term': 'nuclear chromatin'}],\n",
       "     'molecular_function': [{'code': 'GO:0001228',\n",
       "       'term': 'DNA-binding transcription activator activity, RNA polymerase II-specific'},\n",
       "      {'code': 'GO:0000981',\n",
       "       'term': 'DNA-binding transcription factor activity, RNA polymerase II-specific'},\n",
       "      {'code': 'GO:0000978',\n",
       "       'term': 'RNA polymerase II proximal promoter sequence-specific DNA binding'}]},\n",
       "    'uniprot_keywords': ['Alternative splicing',\n",
       "     'Complete proteome',\n",
       "     'Developmental protein',\n",
       "     'Disease mutation',\n",
       "     'DNA-binding',\n",
       "     'Homeobox',\n",
       "     'Microphthalmia',\n",
       "     'Nucleus',\n",
       "     'Polymorphism',\n",
       "     'Reference proteome',\n",
       "     'Transcription',\n",
       "     'Transcription regulation']}},\n",
       "  'scores': {'association_score': 0.11307470223393612},\n",
       "  'sourceID': 'expression_atlas',\n",
       "  'target': {'activity': 'increased_transcript_level',\n",
       "   'gene_info': {'geneid': 'ENSG00000134438',\n",
       "    'name': 'retina and anterior neural fold homeobox',\n",
       "    'symbol': 'RAX'},\n",
       "   'id': 'ENSG00000134438',\n",
       "   'target_type': 'transcript_evidence'},\n",
       "  'type': 'rna_expression',\n",
       "  'unique_association_fields': {'comparison_name': \"'Crohn's disease' vs 'non inflammatory bowel disease control' in 'A1b Paris classification'\",\n",
       "   'datasource': 'expression_atlas',\n",
       "   'geneID': 'http://identifiers.org/ensembl/ENSG00000134438',\n",
       "   'study_id': 'http://identifiers.org/gxa.expt/E-GEOD-101794'}}}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Show example result\n",
    "es.search(index='master_evidence-data', body={\"query\": {\"match_all\": {}}})['hits']['hits'][0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Setup scanner for entire evidence data index\n",
    "batch_size = 10000\n",
    "query = {\"query\": {\"match_all\": {}}}\n",
    "# 1626542 records as of 20191007\n",
    "res = helpers.scan(es, query, index='master_evidence-data', size=batch_size, scroll='1h')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "! [ -d $export_dir ] && rm -rf $export_dir"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Use fixed schema or resulting parquet dataset will be useless otherwise\n",
    "schema = pa.schema([\n",
    "    pa.field(\"access_level\", pa.string()),\n",
    "    pa.field(\"batch\", pa.int64()),\n",
    "    pa.field(\"disease\", pa.string()),\n",
    "    pa.field(\"drug\", pa.string()),\n",
    "    pa.field(\"evidence\", pa.string()),\n",
    "    pa.field(\"id\", pa.string()),\n",
    "    pa.field(\"literature\", pa.string()),\n",
    "    pa.field(\"private\", pa.string()),\n",
    "    pa.field(\"scores\", pa.string()),\n",
    "    pa.field(\"sourceID\", pa.string()),\n",
    "    pa.field(\"target\", pa.string()),\n",
    "    pa.field(\"type\", pa.string()),\n",
    "    pa.field(\"unique_association_fields\", pa.string()),\n",
    "    pa.field(\"validated_against_schema_version\", pa.string())\n",
    "])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1626543it [17:06, 1584.12it/s]\n"
     ]
    }
   ],
   "source": [
    "def to_value(v):\n",
    "    # Convert objects to json string (leave scalars)\n",
    "    return v if np.isscalar(v) else json.dumps(v)\n",
    "\n",
    "# Loop through batches and write each as a new chunk in the parquet dataset\n",
    "for i, batch in enumerate(more_itertools.chunked(tqdm.tqdm(res), batch_size)):\n",
    "    df = pd.DataFrame([r['_source'] for r in batch])\n",
    "    df = df.applymap(to_value)\n",
    "    df['batch'] = i\n",
    "    df = df[sorted(df.columns.tolist())]\n",
    "    for f in schema:\n",
    "        if f.name not in df:\n",
    "            df[f.name] = None \n",
    "        if f.type.to_pandas_dtype() == np.object_:\n",
    "            df[f.name] = df[f.name].fillna('')\n",
    "        df[f.name] = df[f.name].astype(f.type.to_pandas_dtype())\n",
    "    df = df[[f.name for f in schema]]\n",
    "    df.to_parquet(export_dir, index=False, partition_cols=['batch'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=103\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=104\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=132\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=135\n",
      "12M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=161\n",
      "12M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=159\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=150\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=157\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=134\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=133\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=105\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=102\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=156\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=151\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=158\n",
      "12M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=160\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=31\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=36\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=38\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=53\n",
      "20M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=98\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=54\n",
      "20M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=96\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=62\n",
      "13M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=65\n",
      "8.0M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=91\n",
      "10M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=39\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=37\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=30\n",
      "7.9M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=90\n",
      "12M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=64\n",
      "12M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=63\n",
      "20M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=97\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=55\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=52\n",
      "20M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=99\n",
      "8.0M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=84\n",
      "16M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=70\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=48\n",
      "9.0M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=77\n",
      "8.0M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=83\n",
      "10M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=41\n",
      "8.0M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=79\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=46\n",
      "6.4M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=12\n",
      "6.3M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=15\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=23\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=24\n",
      "8.1M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=78\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=47\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=40\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=49\n",
      "7.9M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=82\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=76\n",
      "15M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=71\n",
      "8.0M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=85\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=25\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=22\n",
      "6.9M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=14\n",
      "6.7M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=13\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=142\n",
      "7.3M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=5\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=145\n",
      "7.4M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=2\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=120\n",
      "20M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=118\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=127\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=111\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=129\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=116\n",
      "16M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=144\n",
      "7.2M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=3\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=143\n",
      "7.3M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=4\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=128\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=117\n",
      "20M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=110\n",
      "20M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=119\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=126\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=121\n",
      "6.9M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=162\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=154\n",
      "9.9M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=153\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=107\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=138\n",
      "20M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=100\n",
      "17M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=136\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=109\n",
      "17M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=131\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=152\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=155\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=130\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=137\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=108\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=101\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=106\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=139\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=57\n",
      "13M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=68\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=50\n",
      "13M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=66\n",
      "7.9M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=92\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=59\n",
      "17M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=95\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=61\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=35\n",
      "9.8M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=32\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=60\n",
      "13M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=94\n",
      "8.5M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=93\n",
      "13M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=67\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=58\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=51\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=56\n",
      "14M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=69\n",
      "10M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=33\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=34\n",
      "7.2M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=16\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=29\n",
      "7.0M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=11\n",
      "9.9M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=27\n",
      "6.7M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=18\n",
      "9.7M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=20\n",
      "12M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=74\n",
      "7.9M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=80\n",
      "7.7M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=87\n",
      "12M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=73\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=45\n",
      "8.0M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=89\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=42\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=21\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=26\n",
      "6.5M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=19\n",
      "7.3M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=10\n",
      "7.2M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=17\n",
      "9.8M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=28\n",
      "8.0M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=88\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=43\n",
      "11M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=44\n",
      "13M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=72\n",
      "7.7M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=86\n",
      "8.0M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=81\n",
      "13M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=75\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=124\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=123\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=115\n",
      "17M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=112\n",
      "7.5M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=146\n",
      "7.0M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=1\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=141\n",
      "7.1M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=6\n",
      "7.7M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=148\n",
      "7.1M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=8\n",
      "17M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=113\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=114\n",
      "18M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=122\n",
      "19M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=125\n",
      "6.1M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=9\n",
      "7.8M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=149\n",
      "20M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=140\n",
      "7.2M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=7\n",
      "7.0M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=147\n",
      "6.5M\t/lab/data/raw/evidence-data-stage-2.parquet/batch=0\n",
      "2.0G\t/lab/data/raw/evidence-data-stage-2.parquet\n",
      "2.0G\ttotal\n"
     ]
    }
   ],
   "source": [
    "!du -ch $export_dir"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
